designs/closures-design.md

```diff
@@ -0,0 +1,362 @@
+# Closures Design Document
+
+## What we implemented
+
+We implemented what we committed to with the exception of free and nonlocal variables. We 
+mostly stuck to our original plan.
+
+If expressions are fully implemented, however once inheritance is implemented
+more work may need to be done.
+
+First class functions are implemented except for methods.
```

Comment:
I don't understand what "except for methods" means here.

Reply:
We don't currently support returning or storing references to methods (specifically, class methods), just top-level or nested functions. We'd have to think about the calling syntax and representation of bound/unbound methods, e.g. `f = Cls.func` vs. `f = Cls().func`.